src/common/k8s-api/endpoints/role.api.ts

```diff
@@ -4,11 +4,10 @@
  */
 
 import { KubeObject } from "../kube-object";
-import { KubeApi } from "../kube-api";
-import { isClusterPageContext } from "../../utils/cluster-id-url-parsing";
+import { KubeApi, SpecificApiOptions } from "../kube-api";
 
 export interface Role {
-  rules: {
+  rules?: {
```

Comment:
Hmm. In general, perhaps we tolerate nullable types a bit too much in the codebase, as they contribute to complexity quite a bit. Just a thought.

Reply:
I agree, however we must respect the fact that a lot of these are because they are nullable from kube itself.